src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,265 @@
+                } else
+                {
+                    fileNameExpression = name.Add(Literal(".wav"));
+                }
+                if (valueSerialization.ContentType != "application/json")
+                {
+                    contentTypeExpression = Literal(valueSerialization.ContentType);
+                }
+            }
+            if (valueSerialization.Type.IsFrameworkType)
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+            }
+            if (contentExpression != null)
+            {
+                return Configuration.ApiTypes.GetMultipartFormDataRequestContentAddStatment(mulitpartContent, contentExpression, name, fileNameExpression, contentTypeExpression);
+            }
+
+            switch (valueSerialization.Type.Implementation)
+            {
+                case SerializableObjectType serializableObjectType:
+                    contentExpression = BuildValueSerizationExpression(serializableObjectType.Type, valueExpression);
+                    break;
+                case ObjectType:
+                    contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                    break;
+                case EnumType { IsIntValueType: true, IsExtensible: false } enumType:
+                    contentExpression = BuildValueSerizationExpression(typeof(int), new CastExpression(valueExpression, enumType.ValueType));
+                    break;
+                case EnumType { IsNumericValueType: true } enumType:
+                    contentExpression = BuildValueSerizationExpression(typeof(float), new EnumExpression(enumType, valueExpression).ToSerial());
+                    break;
+                case EnumType enumType:
+                    contentExpression = BuildValueSerizationExpression(typeof(string), new EnumExpression(enumType, valueExpression).ToSerial());
+                    break;
+                default:
+                    throw new NotSupportedException($"Cannot build serialization expression for type {valueSerialization.Type}, please add `CodeGenMemberSerializationHooks` to specify the serialization of this type with the customized property");
+            }
+            return Configuration.ApiTypes.GetMultipartFormDataRequestContentAddStatment(mulitpartContent, contentExpression, name, fileNameExpression, contentTypeExpression);
+        }
+
+        private static ValueExpression BuildValueSerizationExpression(CSharpType valueType,  ValueExpression valueExpression) => valueType switch
+        {
+            _ when valueType.IsFrameworkType => valueExpression,
+            _ when valueType.Implementation is SerializableObjectType serializableObjectType => new InvokeStaticMethodExpression(typeof(ModelReaderWriter), nameof(ModelReaderWriter.Write), new[] { valueExpression, ModelReaderWriterOptionsExpression.Wire }, new[] { valueType }),
```

Comment:
we should use pattern matching:
```suggestion
            { IsFrameworkType: true } => valueExpression,
            { IsFrameworkType: false, Implementation: SerializableObjectType serializableObjectType } => ModelReaderWriterExpression.Write(valueExpression, ModelReaderWriterOptionsExpression.Wire),
```

Reply:
also why "wire" format here?

Reply:
When the property is a Model, we need to serialize it as serialize in wire.